### PR TITLE
Address recent drop in usage data because usagelevel set to off

### PR DESCRIPTION
**Pull request by TCourtneyOwen**
- The fix is to ensure office-addin-scripts packages don't write out the office-addin-usage-data file before Yo Office does
- Created an array of office-addin-scripts packages currently consuming office-addin-usage-data package and check if the project name is contained in that array during object construction.  If it is, then don't write out the office-addin-usage-data file.
- Also fixed some formatting issues

Yo office imports other packages such as office-addin-manifest, which when loaded as part of an import, invoke the following to create a new usageData object:
export const usageDataObject: usageData.OfficeAddinUsageData = new usageData.OfficeAddinUsageData({
    projectName: "office-addin-manifest",
    instrumentationKey:  usageData.instrumentationKeyForOfficeAddinCLITools,
    raisePrompt: false
});

In the constructor for the office-addin-usage-data, we have the following bit of code:

      if (!this.options.isForTesting && this.options.raisePrompt && jsonData.needToPromptForUsageData(this.options.groupName)) {
        this.usageDataOptIn();
      } else {
        jsonData.writeUsageDataJsonData(this.options.groupName, this.options.usageDataLevel);
      }
Because raisePrompt is set to false, we end up calling jsonData.writeUsageDataJsonData with the usageDataLevel turned to off, and write out the office-addin-usage-data file with the setting. Once this file is written, yo office will see it's written and use these settings moving forward.

### CODE REVIEW

**Comment by akrantz** [comment]
@TCourtneyOwen Wouldn't it be better and simpler to have the default be undefined and to not write anything when undefined so that only the true or false value about whether to prompt would cause the value to be written?

File: packages/office-addin-usage-data/src/defaults.ts [hunk 1]
```diff
@@ -6,4 +6,13 @@ import * as path from "path";
 
 export const usageDataJsonFilePath: string = path.join(os.homedir(), "/office-addin-usage-data.json");
 export const groupName = "office-addin-usage-data";
-export const instrumentationKeyForOfficeAddinCLITools = "de0d9e7c-1f46-4552-bc21-4e43e489a015";
\ No newline at end of file
+export const instrumentationKeyForOfficeAddinCLITools = "de0d9e7c-1f46-4552-bc21-4e43e489a015";
+// Office-Addin-Scripts packages currently sending usage data
+export const officeAddinScriptsPackages: string[] = [
```
Comment (akrantz) [inline packages/office-addin-usage-data/src/defaults.ts:11]: I'm concerned about this pattern where these package names are specified. Isn't there a better way?

**Comment by TCourtneyOwen** [comment]
@akrantz can you please clarify - the default of which variable?

File: packages/office-addin-usage-data/src/defaults.ts (hunk 1 above)
Comment (TCourtneyOwen) [reply packages/office-addin-usage-data/src/defaults.ts:11]: Yeah I am not crazy about this design either. The other option is to update all the consuming office-addin-script packages to not export an usage data object and to just create it at runtime. But this opens up the clunky pattern of creating the usage data object multiple times in each file that sends usage data.  I am really striving for as minimal changes as possible, but if you thinking creating the object at runtime is the best approach, we can do that

Comment (akrantz) [reply packages/office-addin-usage-data/src/defaults.ts:11]: I think a chat might help us discuss and come up with a better proposal to deal with this.

**Comment by TCourtneyOwen** [comment]
Looks like VM deployment failed the past couple days so the CI loop is not running.  Investigating now

**Comment by TCourtneyOwen** [comment]
It appears they renamed the base images in Azure that we use as the foundation for creating our VMs.  I need to update the deployment scripts to use the new name.  I wish they wouldn't change these names willy-nilly

**Pull request merged**